src/core/utils/stringutils.h

```diff
@@ -0,0 +1,46 @@
+/***************************************************************************
+                        stringutils.h
+                        ---------------
+  begin                : Jun 2020
+  copyright            : (C) 2020 by Ivan Ivanov
+  email                : ivan@opengis.ch
+***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef STRINGUTILS_H
+#define STRINGUTILS_H
+
+#include <QObject>
+
+
+class StringUtils : public QObject
+{
+    Q_OBJECT
+
+  public:
+
+
+    explicit StringUtils( QObject *parent = nullptr );
+
+
+    /**
+     * Checks whether the provided string is a relative url (has no protocol or starts with `file://`).
+     */
+    static Q_INVOKABLE bool isRelativeUrl( const QString &url );
+
+
+    /**
+     * Returns a string with any URL (e.g., http(s)/ftp) and mailto: text converted to valid HTML <a …> links.
+     */
+    static Q_INVOKABLE QString insertLinks( const QString &string );
```

Comment:
what about making QGIS one invokable and calling it directly by adding QgsStringUtils with a singleton provider?

Reply:
I have tried three different strategies, all of them failed or decided they are too much:
1) make `QgsStringUtils` a `Q_GADGET` and add via `qmlRegisterUncreatableType`. Then it is available as an object within QML, but has no properties.
2) `REGISTER_SINGLETON` requires `QgsStringUtils` to have a constructor and convert it to `Q_OBJECT`. Isn't it too much for QGIS?
3) `setContextProperty` is similar to 2)

That's why I decided that a simple wrapper can do the job better and all of it is under control in QField. I am open for suggestions to reuse the `QgsStringUtils` directly.